splitio/api/auth.py

```diff
@@ -37,17 +39,21 @@ def authenticate(self):
         :return: Json representation of an authentication.
         :rtype: splitio.models.token.Token
         """
+        start = int(round(time.time() * 1000))
         try:
             response = self._client.get(
                 'auth',
                 '/v2/auth',
                 self._apikey,
-                extra_headers=self._metadata
+                extra_headers=self._metadata,
             )
+            record_telemetry(response.status_code, int(round(time.time() * 1000)) - start, TOKEN, self._telemetry_runtime_producer)
             if 200 <= response.status_code < 300:
                 payload = json.loads(response.body)
                 return from_raw(payload)
             else:
+                if response.status_code == 401:
```

Comment:
What about `if (response.status_code >= 400 && response.status_code < 500)` to cover all 4XX errors? (we are getting 401 for invalid api key, but 400 if using a client-side API Key on server-side)

Reply:
will do.